src/tracer/Tracee.cc

```diff
@@ -41,13 +42,20 @@ Arch Tracee::getSyscallArch() const {
     return syscallArch_;
 }
 
-uint64_t Tracee::getSyscallNumber() {
+reg_t Tracee::getSyscallNumber() {
     if (syscallArch_ == Arch::UNKNOWN)
         throw Exception("Can't get syscall number, unknown syscall arch");
+#if defined(__x86_64__)
     return regs_.orig_rax;
+#elif defined(__i386__)
+    return regs_.orig_eax;
+#else
+#error "arch not supported"
+#endif
 }
 
-uint64_t Tracee::getSyscallArgument(uint8_t argumentNumber) {
+reg_t Tracee::getSyscallArgument(uint8_t argumentNumber) {
+#if defined(__x86_64__)
     if (syscallArch_ == Arch::X86) {
         if (argumentNumber == 0)
             return static_cast<uint32_t>(regs_.rbx);
```

Comment:
To robi cast z `uint64_t`/`int64_t` na `uint32_t`, a potem od razu automatyczną promocję spowrotem na `uint64_t` ?

Reply:
Jakoś tak, intencja była taka, żeby ignorować wyższe bity rejestrów. Nie wiem co w nich jest (skoro program manipuluje tylko połową).

Reply:
Wydaje mi się, że `& 0xFFFFFFF` byłoby czytelniejsze, ale tam jeszcze signedness wchodzi... nie wiem do końca jak to się zachowuje, ale działa ok, więc może lepiej zostawmy jak jest.